manifests/base/application-controller/argocd-application-controller-statefulset.yaml

```diff
@@ -136,12 +136,6 @@ spec:
             port: 8082
           initialDelaySeconds: 5
           periodSeconds: 10
-        livenessProbe:
-          httpGet:
-            path: /healthz
-            port: 8082
-          initialDelaySeconds: 5
-          periodSeconds: 10
```

Comment:
The reasoning about removing it is related to the nature of the controller watch loop versus the how the current liveness probe is currently implemented. Currently the `/healthz` endpoint is exposed by the metrics server which runs in a separate goroutine. This can cause the controller to be restarted by reasons unrelated to the watch loop. In cases when the watch queues are accumulating lots of resource updates the controller will be busy trying to process them. Killing the controller in this case makes the situation worse as the queue remains the same after the restart. We noticed this behaviour in one of our internal instances which led to this discussion challenging the effectiveness of the current liveness probe implementation in ArgoCD controller.

@34fathombelow do you have a real use case where the current liveness-probe implementation is useful?

Reply:
Got it, thanks for your detailed explanation.   I'm fine with this change.

Reply:
Thanks for the extra background info as well @leoluz I was not aware that the check was also in a seperate go routine which makes the liveness prob as you mentioned even more useless.